src/tabs/src/Tab.js

```diff
@@ -59,19 +60,18 @@ const Tab = memo(
         ? { marginRight: '8px' }
         : { marginBottom: '8px' }
 
-    const handleClick = e => {
-      safeInvoke(props.onClick, e)
-      onSelect()
-    }
-
-    const handleKeyPress = e => {
-      if (e.key === 'Enter' || e.key === ' ') {
-        onSelect()
-        e.preventDefault()
-      }
+    const onClickRef = useLatest(props.onClick)
+    const handleClick = useCallback(
+      event => {
+        safeInvoke(onClickRef.current, event)
+        if (!disabled) {
+          onSelect()
+        }
+      },
+      [disabled, onSelect]
+    )
```

Comment:
I had considered a `use-combined-callback` hook, but might be a little overkill for now (unless we see this pattern happening all over the place).